jetbrains-core/src/software/aws/toolkits/jetbrains/core/credentials/profiles/ProfileWatcher.kt

```diff
@@ -43,11 +53,33 @@ class ProfileWatcher(parentDisposable: Disposable) : AsyncFileListener, Disposab
     fun start(onFileChange: () -> Unit) {
         onUpdate = onFileChange
 
-        watchRoots.addAll(LocalFileSystem.getInstance().addRootsToWatch(watchLocationsStrings, false))
+        LOG.info { "Starting profile watcher, profile locations: $watchLocationsStrings" }
+
+        val localFileSystem = LocalFileSystem.getInstance()
+
+        val watchLocationParents = watchLocationsStrings.map {
+            // Make VFS aware of it
+            localFileSystem.refreshAndFindFileByPath(it)
+
+            // Use the parent as the watch root in case file does not exist yet
+            // Note: This system requires that the parent folder already exists
```

Comment:
I don't think we should necessarily solve this, but what does JB do when it doesn't exist? create the file?

Reply:
What do you mean? 

The find call returns null and the VFS ignores it so we won't receive any updates.

Reply:
Right that's what I mean, I think that that is fine, but I guess Intellij typically only calls this after it makes a file

Reply:
Ya